### PR TITLE
Unblock topic

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

File: src/cpp/DataBrokerListener.cpp [hunk 1]
```diff
@@ -53,13 +53,24 @@ bool DataBrokerListener::init(
...
 }
 
+void DataBrokerListener::unlock_topic(
+        const std::string& topic)
+{
+    std::lock_guard<std::recursive_mutex> lock(mutex_);
+    auto it = topics_blocked_.find(topic);
+    if (it != topics_blocked_.end())
+    {
+        topics_blocked_.erase(topic);
```
Comment (jparisu) [inline src/cpp/DataBrokerListener.cpp:70]: erase does not fail if the element does not exist. Could skip the `find` part

File: include/databroker/DataBrokerConfiguration.hpp [hunk 2]
```diff
@@ -77,6 +78,10 @@ struct DataBrokerConfiguration
             DataBrokerConfiguration& configuration,
             const std::string& file_path = DEFAULT_CONFIGURATION_FILE,
             bool verbose = false);
+
+    static bool reload_configuration_file(
+            DataBrokerConfiguration& configuration,
+            const std::string& file_path);
```
Comment (jparisu) [inline include/databroker/DataBrokerConfiguration.hpp:84]: This string could be get from the configuration that is being used.

File: src/cpp/DataBrokerConfiguration.cpp [hunk 3]
```diff
@@ -268,5 +272,38 @@ bool DataBrokerConfiguration::load_configuration_file(
...
 }
 
+bool DataBrokerConfiguration::reload_configuration_file(
+        DataBrokerConfiguration& configuration,
+        const std::string& file_path)
+{
+    YAML::Node config_node;
+
+    logInfo(DATABROKER_CONFIGURATION, "Loading configuration file '" << file_path << "'");
+    configuration.config_file = file_path;
```
Comment (jparisu) [inline src/cpp/DataBrokerConfiguration.cpp:282]: This is probably redundant and could be avoided as the argument to the function

**Review by jparisu — CHANGES_REQUESTED** [review]

File: src/cpp/DataBrokerListener.cpp (hunk 1 above)
Comment (jparisu) [reply src/cpp/DataBrokerListener.cpp:70]: :+1:

File: include/databroker/DataBrokerConfiguration.hpp (hunk 2 above)
Comment (jparisu) [reply include/databroker/DataBrokerConfiguration.hpp:84]: :+1:

**Review by rsanchez15 — COMMENTED** [review]
LGTM

**Review by jparisu — APPROVED** [review]
LGTM

**Pull request merged**